cpp/experiments/employee_factory/source/person.cpp

```diff
@@ -0,0 +1,8 @@
+#include "person.h"
+
+Person::Person(const std::string& name, const std::string& last_name)
```

Comment:
Here, you may have un unnecessary copy of the params. Think of the case where you dont pass in a lvalue reference to this constructor so compiler has to construct a string for you and then copy it into the object members. 

You could instead just take non const non reference string and move it to your member like 
`Constr::Constr(std::string val) : val_{std::move(val)}`

now in case like `Constr a{"string"}` a single instance of std::string is created and moved to your member field. While with const& it would have to copy the data to new string and then copy it again to your member.

Same applies to other container types, if you accept a vector and need to 'consume' it not just read it, just accept a value and move it.

Reply:
I'm researching that topic, I'll update the code when I'm done

Reply:
Done